store/tikv/txn.go

```diff
@@ -69,41 +69,47 @@ type KVTxn struct {
 	commitCallback func(info kv.TxnInfo, err error)
 }
 
-func newTiKVTxn(store *KVStore, txnScope string) (*KVTxn, error) {
-	bo := NewBackofferWithVars(context.Background(), tsoMaxBackoff, nil)
-	startTS, err := store.getTimestampWithRetry(bo, txnScope)
-	if err != nil {
-		return nil, errors.Trace(err)
+func extractStartTs(store *KVStore, options kv.TransactionOption) (uint64, error) {
+	var startTs uint64
+	var err error
+	if options.StartTS != nil {
+		if options.PrevSec != nil {
+			panic("StartTS and PrevSec should not be set at the same")
+		}
+		startTs = *options.StartTS
+	} else if options.PrevSec != nil {
+		bo := NewBackofferWithVars(context.Background(), tsoMaxBackoff, nil)
+		startTs, err = store.getStalenessTimestamp(bo, options.TxnScope, *options.PrevSec)
+	} else {
+		bo := NewBackofferWithVars(context.Background(), tsoMaxBackoff, nil)
+		startTs, err = store.getTimestampWithRetry(bo, options.TxnScope)
 	}
-	return newTiKVTxnWithStartTS(store, txnScope, startTS, store.nextReplicaReadSeed())
+	return startTs, err
 }
 
-// newTiKVTxnWithStartTS creates a txn with startTS.
-func newTiKVTxnWithStartTS(store *KVStore, txnScope string, startTS uint64, replicaReadSeed uint32) (*KVTxn, error) {
-	ver := kv.NewVersion(startTS)
-	snapshot := newTiKVSnapshot(store, ver, replicaReadSeed)
+func newTiKVTxnWithOptions(store *KVStore, options kv.TransactionOption) (*KVTxn, error) {
+	if options.TxnScope == "" {
+		panic("TxnScope unfilled!")
+	}
```

Comment:
Do we have anything like `assert!` in TiDB?

Reply:
If `options.TxnScope` is "", we use `global` instead.